pkg/controller/nodenetworkconfigurationpolicy/conditions/policy.go

```diff
@@ -0,0 +1,166 @@
+		message,
+	)
+}
+
+func setPolicyNotMatching(conditions *nmstatev1alpha1.ConditionList, message string) {
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionProgressing,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+}
+
+func setPolicyFailedToConfigure(conditions *nmstatev1alpha1.ConditionList, message string) {
+	setPolicyFailed(conditions, nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionFailedToConfigure, message)
+}
+
+func setPolicyFailed(conditions *nmstatev1alpha1.ConditionList, reason nmstatev1alpha1.ConditionReason, message string) {
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionTrue,
+		reason,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionFalse,
+		reason,
+		"",
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionProgressing,
+		corev1.ConditionFalse,
+		reason,
+		"",
+	)
+}
+
+func UpdatePolicyConditions(cli client.Client, policy *nmstatev1alpha1.NodeNetworkConfigurationPolicy) error {
+	logger := log.WithValues("policy", policy.Name)
+	// On conflict we need to re-retrieve enactments since the
+	// conflict can denote that the calculated policy conditions
+	// are now not accurate.
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		enactments := nmstatev1alpha1.NodeNetworkConfigurationEnactmentList{}
+		policyLabelFilter := client.MatchingLabels{nmstatev1alpha1.ENACTMENT_POLICY_LABEL: policy.Name}
+		err := cli.List(context.TODO(), &enactments, policyLabelFilter)
+		if err != nil {
+			return errors.Wrap(err, "geting enactments failed")
+		}
+
+		nodes := corev1.NodeList{}
+		nodeSelectorFilter := client.MatchingLabels(policy.Spec.NodeSelector)
+		err = cli.List(context.TODO(), &nodes, nodeSelectorFilter)
+		if err != nil {
+			return errors.Wrap(err, "geting nodes failed")
+		}
+		numberOfNodesAffected := len(nodes.Items)
+		numberOfEnactments := len(enactments.Items)
+
+		// Let's get conditions with true status frequency
+		enactmentsCount := enactment.Count(enactments)
+
+		logger.Info(fmt.Sprintf("enactments count: %s", enactmentsCount.String()))
+
+		if enactmentsCount.Failed() > 0 {
+			message := fmt.Sprintf("%d/%d nodes failed to configure", enactmentsCount.Failed(), numberOfNodesAffected)
+			setPolicyFailedToConfigure(&policy.Status.Conditions, message)
+
+		} else if enactmentsCount.Progressing() > 0 {
```

Comment:
I think this is redundant. We don't care if any enactment is "Progressing". As long as not all Nodes have Enactment and not all Enactments are either Degraded or Available, the Policy is progressing. Plus, I think that the information on how many nodes it is Available, Degraded and YetToMatch is just as interesting for the user to know.